R/epidemic.R

```diff
@@ -0,0 +1,26 @@
+time_to_peak <- function(df, dt = 1) {
+  df |>
+    filter(state == "I") |>
+    slice_max(order_by = proportion, n = 1, with_ties = FALSE) |>
+    rename(time_to_peak = dt * timestep) |>
+    pull(time_to_peak)
+}
+
+epidemic_final_size <- function(df) {
+  df |>
+    filter(state %in% c("E", "I", "R")) |>
+    slice_max(order_by = timestep, n = 1, with_ties = TRUE) |>
```

Comment:
@athowes does this select the maximum value in each of `E`, `I` and `R` or the final value?

I think it does the former which isn't what we want (the maximum values of `E`, `I` and `R` will occur at different times and will involve the same people who, over the course of a model simulation, transition between `E` -> `I` -> `R`).